antsibull/schemas/base.py

```diff
@@ -134,7 +134,7 @@ def handle_renamed_attribute(cls, values):
 
 #: option types are a set of strings that represent the types handled by argspec.
 OPTION_TYPE_F = p.Field('str', regex='^(bits|bool|bytes|dict|float|int|json|jsonarg|list'
-                        '|path|raw|sid|str)$')
+                        '|path|raw|sid|str|tmp|temppath|tmppath|pathspec|pathlist)$')
```

Comment:
I don't think we want to have multiple things with the same meaning.  `temppath` and `tmppath` definitely seem to fall under that.  I think all the rest are different?

What we should do is decide on one of those (I'd use `tmppath` since we're using `tmp` and not `temp`) and then have a validator which converts any fields which are the alternate name into the canonical one.  That way we're liberal in what we accept and strict in what we emit.

Reply:
I don't know exactly why there are these different names (I guess it's historical). The names come from here: https://github.com/ansible/ansible/blob/e6da5443101cc815cb479965ab8d0e81c6d23333/lib/ansible/config/manager.py#L59-L78 According to the documentation there, `tmp`, `tmppath` and `temppath` should be the same.

Normalizing these three to `tmppath` sounds reasonable. I'll implement that.

Reply:
Implemented in a77998155c43ee088d6e33eafc40606eae3567fc.